### ISSUE TITLE
Tile background replaced with white in some cases

### ISSUE THREAD

**Issue by immanuel-h** [mapproxy/mapproxy#657]
## Context
Caching a mostly transparent service from Arcgis, the background of some tiles gets replaced with white. 
Custom grid and source bbox are given as  [1787873, 6058795, 1796055, 6068937], which matches the arcgis extent given. I could observe the behavior both with mapproxy-seed as well as on-demand tile seeding. 

Image returned by arcgis (sorry for gimp screenshot, but otherwise transparency would not be visible of course).
![image](https://user-images.githubusercontent.com/104762949/232498867-bca6dfdb-6ed8-4ada-8fee-b768fb890fcf.png)

Request for this image is: http://.../arcgis/rest/services/[...]/MapServer/export?transparent=True&format=png&bbox=1795726.904656302,6060280.506546704,1796055.0,6060777.347230557&size=275,416&bboxSR=3857&imageSR=3857&f=image

Mapproxy makes this out of it (i.e. everything white):
![image](https://user-images.githubusercontent.com/104762949/232498560-55564186-b4e3-44ce-abaa-8e88e89f8fe8.png)

The tile to the left looks fine though, as we can see here on the demo service:
![image](https://user-images.githubusercontent.com/104762949/232500512-4333c56c-40bd-4544-8063-62eff63d754f.png)

## Expected Behavior
Whole background should remain transparent. 

## Actual Behavior
Part of the background, mainly on the right and lower part, are replaced with white. This seems to affect tiles that are partially outside of the extent of the map itself. 

## Context
Stops me from shipping this map ... 

## Your Environment
* Version used: Both Mapproxy 1.14.1 and 1.16 show this bug
* Environment name and version (e.g. Python 2.7.5 with mod_wsgi 4.5.9): Python 3.9
* Operating System and version: OpenSuse 15.3

**Comment by gu-LGB** [comment]
We have had a similar problem since switching from 1.13.0 to 1.15.1:

Our service with the following configuration:

```yaml
#-------------------------------------------------------------
#     Caches
#-------------------------------------------------------------

caches:
  webatlasde_bebb_halbton_compressed:
    format: image/png
    image:
      mode: P
      colors: 256
      opacity: 1.0
      transparent: true
      encoding_options:
      # The algorithm used to quantize (reduce) the image colors
        quantizer: fastoctree
    grids: [grid_webatlas_25833_512]
    sources: [webatlas_bebb_wms_farbe]
    link_single_color_images: true
    minimize_meta_requests: true

  topplus_farbe_compressed:
    format: image/png
    image:
      mode: P
      colors: 256
      opacity: 0.2
      transparent: true
      encoding_options:
        quantizer: fastoctree
    grids: [grid_topplus_25833]
    sources: [topplus_open_farbe]
    link_single_color_images: true
    minimize_meta_requests: true

  topplus_farbe_opak:
    disable_storage: true
    format: mixed
    request_format: image/png
    grids: [grid_topplus_25833]
    sources: [topplus_farbe]

  webatlasde_bebb_halbton:
    grids: [grid_webatlas_25833_256]
    sources: [webatlas_bebb_wms_farbe]
    link_single_color_images: true
    minimize_meta_requests: false
    image:
      mode: RGBA
      colors: 0
      transparent: true

  topplus_farbe:
    format: mixed
    request_format: image/png
    grids: [grid_topplus_25833]
    sources: [topplus_open_farbe]
    link_single_color_images: true
    minimize_meta_requests: true
    image:
      mode: RGBA
      colors: 0
      transparent: true
      encoding_options:
        jpeg_quality: 85
      opacity: 0.2

  webatlasde_bebb_grau:
    grids: [grid_webatlas_25833_256]
    sources: [webatlas_bebb_wms_grau]
    link_single_color_images: true
    minimize_meta_requests: false
    image:
      mode: RGBA
      colors: 0
      transparent: true

  # Topplus Graustufen Cache ist nicht zusaetzlich physisch abgelegt, vorhandener topplus_farbe ist eingefaerbt
  topplus_grau:
    disable_storage: true
    format: mixed
    request_format: image/png
    grids: [grid_topplus_25833]
    link_single_color_images: true
    minimize_meta_requests: true
    image:
      mode: RGBA
      colors: 0
      transparent: true
      opacity: 0.2
    sources:
      l: [{source: topplus_farbe, band: 0, factor: 0.21},
          {source: topplus_farbe, band: 1, factor: 0.72},
          {source: topplus_farbe, band: 2, factor: 0.07}
      ]

#-------------------------------------------------------------
#     Sources
#-------------------------------------------------------------

sources:
  webatlas_bebb_wms_farbe:
    type: wms
    seed_only: true
    concurrent_requests: 16
    wms_opts:
      version: 1.3.0
      featureinfo: false
      legendurl: https://isk.lgb.lvnbb.de/ows/legends/Legende_WebAtlasDE_halbton.png
    image:
      format: image/png
      opacity: 1.0
    req:
      url: https://aaa-dienste-test.lgb.local/aaa-suite/cgi-bin/webatlas/wms?
      layers: WebAtlasDE
      styles: palette_rgb_halbton
      transparent: true

  webatlas_bebb_wms_grau:
    type: wms
    seed_only: true
    concurrent_requests: 16
    wms_opts:
      version: 1.3.0
      featureinfo: false
      legendurl: https://isk.lgb.lvnbb.de/ows/legends/Legende_WebAtlasDE_grau.png
    image:
      format: image/png
      opacity: 1.0
    req:
      url: https://aaa-dienste-test.lgb.local/aaa-suite/cgi-bin/webatlas/wms?
      layers: WebAtlasDE
      styles: palette_grau
      transparent: true

  topplus_open_farbe:
    type: wms
    seed_only: true
    concurrent_requests: 16
    wms_opts:
      version: 1.3.0
      featureinfo: false
      legendgraphic: true
    image:
      format: image/png
      opacity: 0.35
    req:
      url: https://sg.geodatenzentrum.de/wms_topplus_open
      layers: web_scale
      transparent: true

#-------------------------------------------------------------
#     Grids
#-------------------------------------------------------------

grids:
  default_grid:
    tile_size: [256, 256]
    origin: ul

  grid_webatlas_25833_256:
    base: default_grid
    srs: 'EPSG:25833'
    bbox_srs: 'EPSG:25833'
    bbox: [245149, 5685471, 500600, 5940137]
    res: [
       #Res                    # AdV-Level  LGB-Level  Scale (96 dpi)
       #4891.96981025128000000,     # 0            17.471.320.7508974
       2445.98490512564000000,      # 1      0      8.735.660.37544871
       1222.99245256282000000,      # 2      1      4.367.830.18772436
        611.49622628141000000,      # 3      2      2.183.915.09386218
        305.74811314070500000,      # 4      3      1.091.957.54693109
        152.87405657035300000,      # 5      4       545.978.773465545
         76.43702828517630000,      # 6      5       272.989.386732772
         38.21851414258810000,      # 7      6       136.494.693366386
         28,                        #        7       100.000
         19.10925707129410000,      # 8      8        68.247.3466831931
         14,                        #        9        50.000
          9.55462853564703000,      # 9      10       34.123.6733415964
          7,                        #        11       25.000
          4.77731426782352000,      # 10     12       17.061.8366707983
          2.8,                      #        13       10.000
          2.38865713391176000,      # 11     14        8.530.91833539913
          1.96,                     #        15        7.000
          1.4,                      #        16        5.000
          1.19432856695588000,      # 12     17        4.265.45916769957
          0.84,                     #        18        3.000
          0.59716428347793900,      # 13     19        2.132.72958384978
          0.42,                     #        20        1.500
          0.28,                     #        21        1.000
          0.14                      #        22         500
    ]

  grid_webatlas_25833_512:
    tile_size: [512, 512]
    origin: ul
    srs: 'EPSG:25833'
    bbox_srs: 'EPSG:25833'
    bbox: [245149, 5685471, 500600, 5940137]
    res: [
       # Res                   # AdV-Level  LGB-Level  Scale (96 dpi)
       # 4891.96981025128000000,    # 0            17471320.7508974
       2445.98490512564000000,      # 1      0      8735660.37544871
       1222.99245256282000000,      # 2      1      4367830.18772436
        611.49622628141000000,      # 3      2      2183915.09386218
        305.74811314070500000,      # 4      3      1091957.54693109
        152.87405657035300000,      # 5      4       545978.773465545
         76.43702828517630000,      # 6      5       272989.386732772
         38.21851414258810000,      # 7      6       136494.693366386
         28,                        #        7       100000
         19.10925707129410000,      # 8      8        68247.3466831931
         14,                        #        9        50000
          9.55462853564703000,      # 9      10       34123.6733415964
          7,                        #        11       25000
          4.77731426782352000,      # 10     12       17061.8366707983
          2.8,                      #        13       10000
          2.38865713391176000,      # 11     14        8530.91833539913
          1.96,                     #        15        7000
          1.4,                      #        16        5000
          1.19432856695588000,      # 12     17        4265.45916769957
          0.84,                     #        18        3000
          0.59716428347793900,      # 13     19        2132.72958384978
          0.42,                     #        20        1500
          0.28,                     #        21        1000
          0.14                      #        22         500
          ]

  grid_topplus_25833:
    tile_size: [512, 512]
    origin: ul
    srs: 'EPSG:25833'
    bbox_srs: 'EPSG:25833'
    bbox: [-392421, 5099820, 835247, 6286040]
    res: [
    #   Res                   # AdV-Level  LGB-Level  Scale (96 dpi)
    #  4891.96981025128000000,      # 0             17471320.7508974
       2445.98490512564000000,      # 1      0      8.735.660.37544871
       1222.99245256282000000,      # 2      1      4.367.830.18772436
        611.49622628141000000,      # 3      2      2.183.915.09386218
        305.74811314070500000,      # 4      3      1.091.957.54693109
        152.87405657035300000,      # 5      4       545.978.773465545
         76.43702828517630000,      # 6      5       272.989.386732772
         38.21851414258810000,      # 7      6       136.494.693366386
         28,                        #        7       100.000
         19.10925707129410000,      # 8      8        68247.3466831931
         14,                        #        9        50000
          9.55462853564703000,      # 9      10       34123.6733415964
          7,                        #        11       25000
          4.77731426782352000,      # 10     12       17061.8366707983
          2.8                        #        13       10000
    ]

#-------------------------------------------------------------
#     Globals
#-------------------------------------------------------------

globals:
  srs:
    proj_data_dir: '/etc/mapproxy/conf/epsg'
    axis_order_ne: ['EPSG:4839', 'EPSG:3045', 'EPSG:3035', 'EPSG:3034']
    axis_order_en: []
  cache:
    concurrent_tile_creators: 8
    meta_size: [8, 8]
    meta_buffer: 340
  http:
    client_timeout: 1000

```

shows the following behavior with the different versions:

**v1.13.0**
![wms](https://github.com/mapproxy/mapproxy/assets/6780678/9d8297b0-d1c1-4ebe-b419-14890e39c863)

**v1.15.1**
![wms](https://github.com/mapproxy/mapproxy/assets/6780678/6ab6252f-8dbc-4ce1-bf00-cd5e90cff043)

**Here is the request (v1.15.1)**
https://isk.geobasis-bb.de/mapproxy/webatlasde_topplus/service/wms?SERVICE=WMS&VERSION=1.1.1&REQUEST=GetMap&FORMAT=image%2Fpng&TRANSPARENT=false&CACHEID=4827514&LAYERS=topplus_farbe%2Cwebatlas_farbe&WIDTH=512&HEIGHT=512&SRS=EPSG%3A25833&STYLES=&BBOX=442800%2C5809000%2C657840%2C6024040

However, the problem does not occur at every zoom level.

**Environment**

    Version used: Mapproxy 1.15.1 -> https://packages.debian.org/bookworm/mapproxy
    Environment name and version: Python 3.11.2 & mod_wsgi 4.9.4
    Operating System and version: Debian 12 (Bookworm)

**Comment by hwbllmnn** [comment]
For file caches this problem is fixed with #828

**Comment by weskamm** [comment]
closed by #828
